pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -422,6 +425,19 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []v1.Volume, namespace s
 				continue
 			}
 
+			scName := pvc.Spec.StorageClassName
+			storageClass, err := c.scInfo.GetStorageClassInfo(*scName)
+			if err != nil {
+				klog.V(4).Infof("Unable to look up StorageClass info for %s/%s, assuming PVC matches predicate when counting limits: %v", namespace, pvcName, err)
+				filteredVolumes[pvID] = true
```

Comment:
I added this line to make it consistent with line 424 above, but IMO the volume shouldn't be counted here (nor if the PVC is not found, like it's done above).

I think it should only be counted after line 436 below, because until then there's no way to know that the volume belongs to the running predicate (that can be GCE, AWS, Azure, Cinder).

@jsafrane, can you take a look at this?

Reply:
> Can we make pods with unbound PVCs with missing storage class unschedulable?

We can return the error and make the pod unschedulable in the node. 

This might be a problem after upgrade, e.g., when the node has a pod running whose volume has no SC, no new pods will be scheduled to it I guess.